modules/apps/27-interchain-accounts/keeper/relay.go

```diff
@@ -0,0 +1,252 @@
+package keeper
+
+import (
+	"encoding/binary"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/ibc-go/modules/apps/27-interchain-accounts/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/modules/core/24-host"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+)
+
+// TODO: implement middleware functionality, this will allow us to use capabilities to
+// manage helper module access to owner addresses they do not have capabilities for
+func (k Keeper) TrySendTx(ctx sdk.Context, portID string, data interface{}) ([]byte, error) {
+	// Check for the active channel
+	activeChannelId, found := k.GetActiveChannel(ctx, portID)
+	if !found {
+		return nil, types.ErrActiveChannelNotFound
+	}
+
+	sourceChannelEnd, found := k.channelKeeper.GetChannel(ctx, portID, activeChannelId)
+	if !found {
+		return []byte{}, sdkerrors.Wrap(channeltypes.ErrChannelNotFound, activeChannelId)
+	}
+
+	destinationPort := sourceChannelEnd.GetCounterparty().GetPortID()
+	destinationChannel := sourceChannelEnd.GetCounterparty().GetChannelID()
+
+	return k.createOutgoingPacket(ctx, portID, activeChannelId, destinationPort, destinationChannel, data)
+}
+
+func (k Keeper) createOutgoingPacket(
+	ctx sdk.Context,
+	sourcePort,
+	sourceChannel,
+	destinationPort,
+	destinationChannel string,
+	data interface{},
+) ([]byte, error) {
+
+	if data == nil {
+		return []byte{}, types.ErrInvalidOutgoingData
+	}
+
+	var msgs []sdk.Msg
+
+	switch data := data.(type) {
+	case []sdk.Msg:
+		msgs = data
+	case sdk.Msg:
+		msgs = []sdk.Msg{data}
+	default:
+		return []byte{}, types.ErrInvalidOutgoingData
+	}
```

Comment:
Thoughts on just requiring the data to be `[]sdk.Msg`?  I don't see much benefit in just allowing sdk.Msg, so it might be nicer to keep the code a little more strict/clean. @seantking ?

Reply:
Yeah, maybe a good idea. I will think about this a bit today :thinking: